tests/test_py.py

```diff
@@ -2841,6 +2843,18 @@ def test_func(x: "Path") -> None:
     assert "BeartypeCallHintForwardRefException" not in result
     assert "beartypeok" in result
 
+@pytest.mark.skipif(shutil.which("gdb") is None, reason="gdb is required for this test")
+def test_inject_insufficient_permissions():
+    """Test that having insufficient permissions for gdb to attach triggers an error."""
+    child = subprocess.Popen(["python", "-c", "import time; time.sleep(20)"])
+
+    with open("/proc/sys/kernel/yama/ptrace_scope") as f:
+        if f.read().strip() == "0":
+            pytest.skip(msg="ptrace is allowed without elevated user permissions")
+
+    # Should fail due to permissions
+    with pytest.raises(Exception):
+        inject(child.pid, [])
```

Comment:
That will fail on macos if gdb installed; I've pushed a commit that also skip on macos